internal/testutil/replay/hoverfly_setup.go

```diff
@@ -0,0 +1,124 @@
+package replay
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"path/filepath"
+	"testing"
+
+	hoverfly "github.com/SpectoLabs/hoverfly/core"
+	v2 "github.com/SpectoLabs/hoverfly/core/handlers/v2"
+)
+
+const simulationDir = "../../../simulations/"
+
+func IsInCaptureMode() bool {
+	return os.Getenv("REPLAY_MODE") == "capture"
+}
+
+func IsInSimulateMode() bool {
+	return os.Getenv("REPLAY_MODE") == "simulate"
+}
+
+func SetupReplayProxy(t *testing.T) (proxyPort *int, teardown func(t *testing.T)) {
+	t.Helper()
+	if IsInCaptureMode() {
+		return setupCaptureMode()
+	}
+	if IsInSimulateMode() {
+		return setupSimulateMode(t)
+	}
+
+	log.Printf("No replay mode was configured")
+	return nil, func(t *testing.T) {
+		t.Helper()
+	}
+}
+
+func setupSimulateMode(t *testing.T) (proxyPort *int, teardown func(t *testing.T)) {
+	t.Helper()
+	port, hv := newHoverflyInstance()
+
+	fileName := fmt.Sprintf("%s%s.json", simulationDir, t.Name())
+	if err := hv.ImportFromDisk(fileName); err != nil {
+		log.Fatalf("Failed to import simulation for test: %v", err)
+	}
+
+	if err := hv.SetMode("simulate"); err != nil {
+		log.Fatalf("Failed to set Hoverfly mode: %v", err)
+	}
+
+	return &port, teardownSimulate(hv)
+}
+
+func setupCaptureMode() (proxyPort *int, teardown func(t *testing.T)) {
+	port, hv := newHoverflyInstance()
+
+	if err := hv.SetModeWithArguments(v2.ModeView{Mode: "capture", Arguments: v2.ModeArgumentsView{Stateful: true}}); err != nil {
+		log.Fatalf("Failed to set Hoverfly mode: %v", err)
+	}
+	return &port, teardownCapture(hv)
+}
+
+func newHoverflyInstance() (int, *hoverfly.Hoverfly) {
+	var min int64 = 1024
+	var max int64 = 65536
+	diff := max - min
+	nBig, err := rand.Int(rand.Reader, big.NewInt(diff+1))
+	if err != nil {
+		log.Fatalf("Failed to generate random number: %v", err)
+	}
+	proxyPort := int(nBig.Int64() + min)
+
+	settings := hoverfly.InitSettings()
+	settings.ProxyPort = fmt.Sprintf("%d", proxyPort)
+	hv := hoverfly.NewHoverflyWithConfiguration(settings)
+
+	if err := hv.StartProxy(); err != nil {
```

Comment:
I think no changes needed for this PR. if we ran this in CI in many resources there will be more chances of picking an already-used port so we might want to implement a retry mechanism in the future to try in a different port until it finds one that is free